src/Task/Route.pm

```diff
@@ -318,7 +318,7 @@ sub iterate {
 				# If it is, then we've moved to an unexpected place. This could be caused by auto-attack,
 				# for example.
 				my %nextPos = (x => $self->{new_x}, y => $self->{new_y});
-				if (distance(\%nextPos, $pos) > $config{$self->{actor}{configPrefix}.'route_step'}) {
+				if (distance(\%nextPos, $pos) > ($config{$self->{actor}{configPrefix}.'route_step'} * 2)) {
```

Comment:
This change should be reviewed, the same check is also made in maproute and seems to be able to cause some errors

Reply:
now, i think i understand why my old bot frequently stuck while on stairs
almost arrived on destination on stair , but move to different floor to attack monster
after monster is killed, stuck, 
because , although very short distance but on different floor, the LOS from last point should check too?